azure/scope/identity.go

```diff
@@ -196,12 +184,31 @@ func (p *AzureCredentialsProvider) GetTokenCredential(ctx context.Context, resou
 		}
 		cred, authErr = azidentity.NewClientSecretCredential(p.GetTenantID(), p.Identity.Spec.ClientID, clientSecret, &options)
 
+	case infrav1.ServicePrincipalCertificate:
+		log.Info("Identity type ServicePrincipalCertificate is deprecated and will be removed in a future release. See https://capz.sigs.k8s.io/topics/workload-identity to get started with Workload Identity instead.")
```

Comment:
I think we might  want to "undeprecate" `ServicePrincipalCertificate` and `UserAssignedMSI` since they no longer use AAD pod identity, wdyt @nojnhuh @mboersma ?

Reply:
I don't have a strong opinion. I like the idea of supporting fewer mechanisms for auth, but flexibility is better for users. And it's a bit odd to un-deprecate things, but AFAICT the deprecation was driven by AAD Pod Identity only.

Reply:
I updated the code and docs so only `ManualServicePrincipal` is deprecated (because it's redundant with `ServicePrincipal`).